cpp/models/ode_seir/model.h

```diff
@@ -108,8 +108,33 @@ class Model : public CompartmentalModel<InfectionState, Populations<InfectionSta
         return temp;
     }
 
+    /**
+    * @brief interpolate reproduction numbers at freely chosen time value that lies in between the time points of the given time series.
+    * values are linearly interpolated from their immediate neighbors from the old time points.
+    * @param y TimeSeries whose reproduction numbers we want to interpolate
+    * @param t_value double time at which we approximate the reproduction number
+    * @return interpolated reproduction number at given time value
+    */
+    IOResult<double> interpolate_reproduction_numbers(const mio::TimeSeries<ScalarType>& y, const double t_value)
+    {
+        if (t_value < -1 || t_value > y.get_num_time_points()) {
+            return mio::failure(
+                mio::StatusCode::OutOfRange,
+                "Cannot interpolate reproduction number too far outside computed horizon of the TimeSeries");
+        }
+        if (t_value < 0) {
+            return get_reproduction_number(0, y);
+        }
+        if (t_value > y.get_num_time_points() - 1) {
+            return get_reproduction_number(y.get_num_time_points() - 1, y);
+        }
+        double y1 = get_reproduction_number(floor(t_value), y).value();
+        double y2 = get_reproduction_number(ceil(t_value), y).value();
```

Comment:
U Are assuming, that the time points are all ints. Which is only the case, when we interpolated the time series before. Otherwise, the time points are choosen from the numerical integrator. 
This should be solved indepently.